extensions/amp-date-display/0.2/test/test-date-display.js

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {DateDisplay} from '../date-display';
+import {createElement} from '../../../../src/preact';
+import {mount} from 'enzyme';
+
+describes.sandboxed('date-display preact component', {}, env => {
+  let sandbox;
+
+  function render(data) {
+    return JSON.stringify(
+      data,
+      (key, value) => {
+        if (typeof value === 'number') {
+          return String(value);
+        }
+        return value;
+      },
+      4
+    );
+  }
+
+  beforeEach(() => {
+    sandbox = env.sandbox;
+    sandbox.useFakeTimers(new Date('2018-01-01T08:00:00Z'));
+  });
+
+  // Unfortunately, we cannot test the most interesting case of UTC datetime
+  // displayed in local, because the test would work in only one time zone.
+
+  it('provides all variables in UTC and English (default)', () => {
+    const props = {
```

Comment:
Why not create a helper method that takes `props` and returns `data` for all these test cases?

Reply:
go/tott/598. Basically, don't rely on test logic when it's not necessary.